tools/azsdk-cli/Azure.Sdk.Tools.Cli/Configuration/RequirementsV1.json

```diff
@@ -0,0 +1,121 @@
+{
+    "categories": {
+        "core": [
+            {   
+                "requirement": "Node.js >= 22.16.0",
+                "check": ["node", "--version"],
+                "instructions": ["Download Node.js"]
+            },
+            {
+                "requirement": "tsp-client >= 0.24.0",
+                "check": ["tsp-client", "--version"],
+                "instructions": ["cd eng/common/tsp-client", "`npm ci`"]
+            },
+            {
+                "requirement": "tsp >= 1.0.0",
+                "check": ["tsp", "--version"],
+                "instructions": ["npm install -g @azure-tools/typespec-client-generator-cli@latest @typespec/compiler@latest"]
+            },
+            {
+                "requirement": "Powershell >= 7",
+                "check": ["pwsh", "--version"],
+                "instructions": ["Download and install PowerShell"]
+            },
+            {
+                "requirement": "github CLI >= 2.30.0",
+                "check": ["gh", "--version"],
+                "instructions": ["Download and install GitHub CLI"]
+            }
+        ],
+        "java": [
+            {
+                "requirement": "java >= 17",
+                "check": ["java", "-version"],
+                "instructions": ["Download JDK", "Set JAVA_HOME environment variable to the JDK installation path", "Add JAVA_HOME/bin to your PATH environment variable", "Restart your IDE."]
+            },
+            {
+                "requirement": "maven",
+                "check": ["mvn", "-v"],
+                "instructions": ["Download the latest version of Maven.", "Set MAVEN_HOME environment variable to the Maven installation path", "Add MAVEN_HOME/bin to your PATH environment variable", "Restart your IDE."]
+            }
+        ],
+        "javascript": [
+            {
+                "requirement": "pnpm",
+                "check": ["pnpm", "--version"],
+                "instructions": ["npm install -g pnpm", "Run pnpm install in the repo root."]
+            }
+        ],
+        "dotnet": [
```

Comment:
I think .NET at some point will need an additional more complicated check for https://learn.microsoft.com/en-us/dotnet/core/deploying/native-aot/?tabs=windows%2Cnet8#prerequisites

Is that the kind of thing that would go in DotNetRequirementsCheck? I still need to do some research about how to validate that prerequisite but I think it might be more work to validate than the other checks represented here